### PR TITLE
Enable comma-dangle rule in ESLint, fix all errors.

### LINKED ISSUE #3304: Trailing comma - enable ESLint rule
It [looks like trailing comma is being prefered](https://github.com/GoogleChrome/lighthouse/pull/3227#discussion_r137820600), however it's not enforced by the current ESLint configuration. Should we [set it up](https://eslint.org/docs/rules/comma-dangle)?

### PR DESCRIPTION
**Pull request by kdzwinel**
Closes #3304

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -38,11 +38,17 @@ module.exports = {
     "strict": [2, "global"],
     "prefer-const": 2,
     "curly": [2, "multi-line"],
+    "comma-dangle": [2, {
+      "arrays": "always-multiline",
+      "objects": "always-multiline",
```
Comment (kdzwinel) [inline .eslintrc.js]: trailing comma is required only if object/array spans to multiple lines

File: .eslintrc.js [hunk 2]
```diff
@@ -38,11 +38,17 @@ module.exports = {
...
+      "imports": "never",
+      "exports": "never",
+      "functions": "never"
```
Comment (kdzwinel) [inline .eslintrc.js]: trailing commas in imports/exports/functions aren't used in lighthouse code, so I turned them off. (`function(a,b,)`?? Yuck)

Comment (patrickhulce) [reply .eslintrc.js]: it's actually pretty nice for clean diffs and easy copy paste of multi-line function arguments :)

```
SomethingVeryLong.SomethingVeryLong = function somethingVeryLong(
  aLongName,
  aSecondLongName,
) {
 //
};
```

Comment (kdzwinel) [reply .eslintrc.js]: Right, with "always-multiline" it would make sense 👍

**Comment by paulirish** [comment]
testing... https://github.com/kdzwinel/lighthouse/blob/a127ad7db074caf35fcda539b06fad37bf721105/lighthouse-cli/test/smokehouse/byte-efficiency/expectations.js#L75-L81

**Pull request merged**